### DISCUSSION TITLE
is it possible to use try catch inside a store

### DISCUSSION THREAD

**Discussion by saif399** [pmndrs/zustand#796] [Q&A]
Hey, 
i'm trying to throw an error inside a function in a store to catch it in a web page

i have this function in a specific page which i want the error to appear here in the catch statement
![image](https://user-images.githubusercontent.com/92998703/150535363-2dd02aba-055f-4c8b-8711-41da7a934582.png)

and i have this inside my store
![image](https://user-images.githubusercontent.com/92998703/150535426-953c51dc-7c9d-491e-9b7b-77c6473e50e0.png)
the inner function throw an error which i want to rethrow to the function in the first image.

the function inside store catches the error but when i try to rethrow it, it doesn't work

any help? Thanks

**Comment by dai-shi** [comment]
The code snippet doesn't tell. It should work. Probably you forgot to put `await` somewhere, because you throw in `async`?

**Reply by jrhager84** [reply]
Doesn't work for me either. I'm asyncing and awaiting with nothing being returned to the invoking function. How does one accomplish this in Zustand?

**Reply by dai-shi** [reply]
https://github.com/pmndrs/zustand/discussions/796#discussioncomment-2155766 should be the answer, which I overlooked in  my first comment.

**Comment by hyphenized** [comment]
> i have this function in a specific page which i want the error to appear here in the catch statement ![image](https://user-images.githubusercontent.com/92998703/150535363-2dd02aba-055f-4c8b-8711-41da7a934582.png)
> 

`updateUserEmail` is async, you need to use `await` to catch any errors that could occur until the promise returned resolves.

```ts
	 try {
      await updateUserEmail(data.currentPassword, data.Email)
    } catch (e: any) {
      console.log(e);
    }
```

**Reply by jrhager84** [reply]
I am using await. I can console log the error in the store, but I can't turn it to the function that called it for error handling. If that makes sense.

**Reply by hyphenized** [reply]
It might have to do something with how you’re passing back the error, if you're using `return` in the block following the catch statement then the function result will be handled as a fulfilled promise, which won't be caught by either `try/catch` or `Promise.catch`.
